### PR TITLE
[Merged by Bors] - feat(FieldTheory/Finite/Basic): `(1 + p * a) ^ (p ^ m) ≡ 1 [MOD p ^ m]`

### PR DESCRIPTION
**Pull request by tb65536**
This PR adds `(1 + p * a) ^ (p ^ m) ≡ 1 [MOD p ^ m]` and deduces that if `n` is a prime power and `a : ZMod n` has order prime to `n`, then either `a = 1` or `a - 1` is a unit. This will be used in the proof that the commutator subgroup of a Z-group is a Hall subgroup.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
These seem unobjectionable.

maintainer merge

File: Mathlib/FieldTheory/Finite/Basic.lean [hunk 1]
```diff
@@ -526,6 +526,41 @@ theorem Int.ModEq.pow_card_sub_one_eq_one {p : ℕ} (hp : Nat.Prime p) {n : ℤ}
...
+      one_mul, Nat.choose_zero_right, Nat.cast_one]
+    refine Nat.ModEq.add_right 1 (Nat.modEq_zero_iff_dvd.mpr ?_)
+    simp_rw [one_pow, mul_one, pow_succ', mul_assoc, ← Finset.mul_sum]
+    refine mul_dvd_mul_left (p ^ m) (dvd_mul_of_dvd_right (Finset.dvd_sum fun k hk ↦ ?_) d)
+    cases m
+    · rw [pow_zero, pow_one, one_mul, add_comm, add_left_inj] at hd
+      cases k <;> simp [← hd, mul_assoc, pow_succ']
+    · cases k <;> simp [mul_assoc, pow_succ']
+
+theorem ZMod.eq_one_or_isUnit {n p k : ℕ} [Fact p.Prime] (hn : n = p ^ k) (a : ZMod n)
```
Comment (jcommelin) [inline Mathlib/FieldTheory/Finite/Basic.lean:545]:
```suggestion
theorem ZMod.eq_one_or_isUnit_sub_one {n p k : ℕ} [Fact p.Prime] (hn : n = p ^ k) (a : ZMod n)
```

**Review by jcommelin — COMMENTED** [review]
bors d+

**Pull request closed without merging**